### PR TITLE
* added HW15

### PR DESCRIPTION
**Pull request by sergkonhar**
@bohdanpc, @helen-black, проверьте пожалуйста домашнее задание #15

### CODE REVIEW

File: Home_Work_15/HW15.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    Student* ptrToBestStudent = findBest(arrayOfStudents, amountOfStudents);
+    std::cout << std::endl <<"Best Student => ";
+    printStudent(*ptrToBestStudent);
+
+    int passedTreshold = sumStudentsPassedTreshold( arrayOfStudents, amountOfStudents, rateTreshold);
+    std::cout << std::endl << "Number of students which pass over 75% avarage rate: " << passedTreshold << std::endl;
+
+}
+
+float avaregeRateOfStudent(Student studentToClaculate)
```
Comment (bohdanpc) [inline Home_Work_15/HW15.cpp]: studentToCalucalte
const Student& , no need to pass the copy just for average calculations

File: Home_Work_15/HW15.cpp [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        avarageRate += studentToClaculate.rates[i];
+    }
+
+    avarageRate /= static_cast<float>(std::size(studentToClaculate.rates));
+
+    return avarageRate;
+}
+
+bool sortByAvarageRate(Student firstStudent, Student secondStudent)
```
Comment (bohdanpc) [inline Home_Work_15/HW15.cpp]: const Student&

File: Home_Work_15/HW15.cpp [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    avarageRate /= static_cast<float>(std::size(studentToClaculate.rates));
+
+    return avarageRate;
+}
+
+bool sortByAvarageRate(Student firstStudent, Student secondStudent) 
+{
+    float avarageRateFirstStudent = 0;
```
Comment (bohdanpc) [inline Home_Work_15/HW15.cpp]: average

File: Home_Work_15/HW15.cpp [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            std::swap(rateSums[i], rateSums[i + 1]);
+            std::swap(arrayOfStudents[i], arrayOfStudents[i + 1]);
+        }
+        
+        else 
+        {
+            bestStudentPtr = (arrayOfStudents + (i+1));
+        }
+    }
+    return bestStudentPtr;
```
Comment (bohdanpc) [inline Home_Work_15/HW15.cpp]: too many steps. Don't even see the need to use additional vector with rates

**Review by bohdanpc — CHANGES_REQUESTED** [review]

File: Home_Work_15/HW15.cpp [hunk 5]
```diff
@@ -29,128 +27,123 @@ int sumStudentsPassedTreshold(Student arrayOfStudents[amountOfStudents], const i
...
-    {
-        avarageRateFirstStudent += firstStudent.rates[i];
-    }
+	for (int i = 0; i < std::size(firstStudent.rates); i++)
+	{
+		avarageRateFirstStudent += firstStudent.rates[i];
+	}
 
-    avarageRateFirstStudent /= static_cast<float>(std::size(firstStudent.rates));
+	avarageRateFirstStudent /= static_cast<float>(std::size(firstStudent.rates));
```
Comment (helen-black) [inline Home_Work_15/HW15.cpp:76]: you can use it:
`float averegeRateOfStudent(const Student& studentToClaculate);`

File: Home_Work_15/HW15.cpp [hunk 6]
```diff
@@ -29,128 +27,123 @@ int sumStudentsPassedTreshold(Student arrayOfStudents[amountOfStudents], const i
...
+	int tempSumFirstStudent = 0;
+	int tempSumSecondStudent = 0;
+	Student* bestStudentPtr = nullptr;
+
+	for (int i = 0; i < amountOfStudents - 1; i++)
+	{
+		for (int j = 0; j < std::size(arrayOfStudents[i].rates); j++)
+		{
+			tempSumFirstStudent += arrayOfStudents[i].rates[j];
+		}
```
Comment (helen-black) [inline Home_Work_15/HW15.cpp:112]: you can use it to compare students
`float averegeRateOfStudent(const Student& studentToClaculate);`

File: Home_Work_15/HW15.cpp [hunk 7]
```diff
@@ -29,128 +27,123 @@ int sumStudentsPassedTreshold(Student arrayOfStudents[amountOfStudents], const i
...
+
+		for (int j = 0; j < std::size(arrayOfStudents[i + 1].rates); j++)
+		{
+			tempSumSecondStudent += arrayOfStudents[i + 1].rates[j + 1];
+		}
+		if (tempSumFirstStudent > tempSumSecondStudent)
+		{
+			bestStudentPtr = &arrayOfStudents[i];
+			std::swap(arrayOfStudents[i], arrayOfStudents[i + 1]);
+		}
```
Comment (helen-black) [inline Home_Work_15/HW15.cpp:122]: you shouldn't change input array, just save the best student

**Review by helen-black — CHANGES_REQUESTED** [review]
it's better to separate meaningfull commits with code style refactoring commits

**Pull request merged**